src/NativeShell/src/cordova/NativeShell.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.cordova.CordovaPlugin;
+import org.apache.cordova.CordovaWebView;
+import org.apache.cordova.PluginResult;
+import org.jellyfin.mobile.R;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.mb.android.media.RemotePlayerService;
+
+import mediabrowser.apiinteraction.android.mediabrowser.Constants;
+
+import static android.view.View.SYSTEM_UI_FLAG_FULLSCREEN;
+import static android.view.View.SYSTEM_UI_FLAG_HIDE_NAVIGATION;
+import static android.view.View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
+import static android.view.View.SYSTEM_UI_FLAG_VISIBLE;
+
+public class NativeShell extends CordovaPlugin {
+    public CallbackContext callbackContext;
+    public JSONArray args;
+
+    public static CordovaWebView cordovaWebView;
+
+    @Override
+    public void pluginInitialize() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    cordova.getActivity().getWindow().setStatusBarColor(cordova.getActivity().getResources().getColor(R.color.gray));
+                    cordova.getActivity().getWindow().setNavigationBarColor(cordova.getActivity().getResources().getColor(R.color.gray));
+                }
+            }
+        });
+    }
+
+    @Override
+    public boolean execute(String action, JSONArray args, final CallbackContext callbackContext) {
+        this.callbackContext = callbackContext;
+        this.args = args;
+
+        cordovaWebView = webView;
+        switch (action) {
+            case "getDeviceInformation":    
+                return getDeviceInformation();
+            case "enableFullscreen":
+                return enableFullscreen();
+            case "disableFullscreen":
+                return disableFullscreen();
+            case "updateMediaSession":
+                return updateMediaSession();
+            case "hideMediaSession":
+                return hideMediaSession();
+            default:
+                callbackContext.error("unrecognized: " + action);
+                return false;
+        }
+    }
+
+    public boolean getDeviceInformation() {
+        // replace this later with a randomly generated persistent string stored in local settings
+        String deviceId =  Settings.Secure.getString(cordova.getActivity().getContentResolver(), Settings.Secure.ANDROID_ID);
+        String deviceName = Build.MODEL;
+        String appName = "Jellyfin Mobile";
+        String appVersion = Integer.toString(BuildConfig.VERSION_CODE);
+
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject.put("deviceId", deviceId);
+            jsonObject.put("deviceName", deviceName);
+            jsonObject.put("appName", appName);
+            jsonObject.put("appVersion", appVersion);
+        } catch(Exception e) {
+            callbackContext.error("error: " + e.getMessage());
```

Comment:
how can an exception be thrown here (filling in json object) but not above? maybe wrap in `try..catch` the whole function, or better the whole `execute()`?